### PR TITLE
Makes cult sacrifice not the end, even if your shade/construct dies.

### PR DESCRIPTION
**Pull request by Arturlang**
## About The Pull Request
Sacrificing a person now only guts or deconstructs(if they are a borg), and allows the soul to return back to the old body once the shade or shell dies.

## Why It's Good For The Game
Getting gibbed and kicked out of the round for good if the construct dies is not all that much fun.
Part of the PRs that should hopefully make cult less quick.

## Changelog
:cl:
tweak: Its now possible to recover from being sacrificed by cult
/:cl:

Also made the shade a construct, to cut down on copypasta.

### CODE REVIEW

**Comment by Arturlang** [comment]
Needs testing

**Comment by Gandalf2k15** [comment]
This is a non-modular change, please follow the modularisation guide.

**Comment by Arturlang** [comment]
It was done during a time where modulirization wasnt neccesary, will redo

**Comment by Arturlang** [comment]
@Gandalf2k15 Check if modularization was done properly, if not what was done incorrectly, first time having to moduralize

**Comment by ForrestWick** [comment]
I don't see a modularization guide anywhere tbh

**Comment by Arturlang** [comment]
@ForrestWick It seems its in the skyrat modular folder as a readme, and the description of the repo has a unset spot for it

File: code/modules/antagonists/cult/runes.dm [hunk 1]
```diff
@@ -202,11 +203,16 @@ structure_check() searches for nearby cultist structures required for the invoca
 		fail_invoke()
 		log_game("Offer rune failed - no eligible targets")
 		return
+	var/mob/living/L = pick(myriad_targets)
+	if(HAS_TRAIT(L, TRAIT_SACRIFICED))
+		fail_invoke()
+		log_game("Offer rune failed - target has already been sacrificed")
+		to_chat(invokers, "<span class='warning'>[L] has already been sacrificed!</span>")
+		return
```
Comment (Gandalf2k15) [inline code/modules/antagonists/cult/runes.dm:211]:
```suggestion
	//SKYRAT EDIT ADDITION BEGIN
	var/mob/living/L = pick(myriad_targets)
	if(HAS_TRAIT(L, TRAIT_SACRIFICED))
		fail_invoke()
		log_game("Offer rune failed - target has already been sacrificed")
		to_chat(invokers, "<span class='warning'>[L] has already been sacrificed!</span>")
		return
		//SKYRAT EDIT END
```

**Review by Gandalf2k15 — CHANGES_REQUESTED** [review]
Have all changes commented like this.

File: code/modules/antagonists/cult/runes.dm (hunk 1 above)
Comment (Arturlang) [reply code/modules/antagonists/cult/runes.dm:211]: Did as requested, did only one comment for one liners, it should be fine given what the documentation for modulirization says

**Comment by Arturlang** [comment]
Needs a testmerge and someone to check if its now properly modularized

File: code/game/objects/items/storage/book.dm [hunk 2]
```diff
@@ -243,7 +243,7 @@ GLOBAL_LIST_INIT(bibleitemstates, list("bible", "koran", "scrapbook", "burning",
 					SS.icon_state = "purified_soulstone2"
 					if(iscultist(M))
 						SSticker.mode.remove_cultist(M.mind, FALSE, FALSE)
-			for(var/mob/living/simple_animal/shade/EX in SS)
+			for(var/mob/living/simple_animal/hostile/construct/shade/EX in SS)
```
Comment (ShadeAware) [inline code/game/objects/items/storage/book.dm]: //SKYRAT EDIT

File: code/game/objects/items/storage/book.dm [hunk 3]
```diff
@@ -253,7 +253,7 @@ GLOBAL_LIST_INIT(bibleitemstates, list("bible", "koran", "scrapbook", "burning",
 		playsound(src,'sound/hallucinations/veryfar_noise.ogg',40,TRUE)
 		if(do_after(user, 40, target = sword))
 			playsound(src,'sound/effects/pray_chaplain.ogg',60,TRUE)
-			for(var/mob/living/simple_animal/shade/S in sword.contents)
+			for(var/mob/living/simple_animal/hostile/construct/shade/S in sword.contents)
```
Comment (ShadeAware) [inline code/game/objects/items/storage/book.dm]: //SKYRAT EDIT

File: code/modules/antagonists/cult/runes.dm [hunk 4]
```diff
@@ -202,11 +202,18 @@ structure_check() searches for nearby cultist structures required for the invoca
...
+		//SKYRAT EDIT END
 	rune_in_use = TRUE
 	visible_message("<span class='warning'>[src] pulses blood red!</span>")
 	var/oldcolor = color
 	color = RUNE_COLOR_DARKRED
-	var/mob/living/L = pick(myriad_targets)
```
Comment (ShadeAware) [inline code/modules/antagonists/cult/runes.dm:209]: Comment this out, don't remove it. Also add //SKYRAT EDIT

File: code/modules/antagonists/cult/runes.dm [hunk 5]
```diff
@@ -303,12 +310,18 @@ structure_check() searches for nearby cultist structures required for the invoca
 		stone.invisibility = 0
 
 	if(sacrificial)
+		//SKYRAT EDIT ADDITION BEGIN
+		ADD_TRAIT(sacrificial, TRAIT_SACRIFICED, "sacrificed")
 		if(iscyborg(sacrificial))
+			var/mob/living/silicon/robot/bot = sacrificial
 			playsound(sacrificial, 'sound/magic/disable_tech.ogg', 100, TRUE)
-			sacrificial.dust() //To prevent the MMI from remaining
```
Comment (ShadeAware) [inline code/modules/antagonists/cult/runes.dm:308]: Comment out, do not delete

File: code/modules/antagonists/cult/runes.dm [hunk 6]
```diff
@@ -303,12 +310,18 @@ structure_check() searches for nearby cultist structures required for the invoca
...
+			bot.deconstruct()
+
 		else
 			playsound(sacrificial, 'sound/magic/disintegrate.ogg', 100, TRUE)
-			sacrificial.gib()
```
Comment (ShadeAware) [inline code/modules/antagonists/cult/runes.dm:311]: Comment out, do not delete

**Review by ShadeAware — CHANGES_REQUESTED** [review]
Needs a read-me file aswell as more _Skyrat Edit_ markers.

**Comment by Arturlang** [comment]
Did I do the readme correctly @ShadeAware

**Comment by Arturlang** [comment]
Travis fail due to unrelated runtime errors

**Comment by Arturlang** [comment]
Ready?

File: code/controllers/subsystem/traumas.dm [hunk 7]
```diff
@@ -46,7 +46,7 @@ SUBSYSTEM_DEF(traumas)
 					   /mob/living/simple_animal/drone, /mob/living/simple_animal/bot, /mob/living/simple_animal/hostile/swarmer)),
 					   "doctors"   = typecacheof(list(/mob/living/simple_animal/bot/medbot)),
 					   "the supernatural"   = typecacheof(list(/mob/living/simple_animal/hostile/construct,
-					   /mob/living/simple_animal/revenant, /mob/living/simple_animal/shade)),
+					   /mob/living/simple_animal/revenant, /mob/living/simple_animal/hostile/construct/shade)), //SKYRAT EDIT, makes shade a construct
```
Comment (Gandalf2k15) [inline code/controllers/subsystem/traumas.dm:49]: Have original... This still doesn't conform to regulations. Please look at any other modularisation compliant PR and take from that.

**Comment by Gandalf2k15** [comment]
Please check up on some of the open PR's on how to properly modularise.

File: code/controllers/subsystem/traumas.dm (hunk 7 above)
Comment (Arturlang) [reply code/controllers/subsystem/traumas.dm:49]: The only PR that appears to mess around with lists is https://github.com/Skyrat-SS13/Skyrat-tg/pull/1383/, and it just adds in the things without any comments or such.
Really unsure what you want me to do, simply telling me to look at open PR's isnt really helpfull.
Do I need to perhaps comment out this list and make a whole new one with the additions as a closed PR does, or += the new thing in a modular spot?

**Comment by Gandalf2k15** [comment]
Its the comments that are wrong, update them to be like other prs. If changing, have the original commented ou4, then the change.

**Comment by Arturlang** [comment]
Unfortunately Ill be unable to maintain this, so ill close this PR

**Pull request closed without merging**